### PR TITLE
Add govulnerability check to bp-cli

### PR DESCRIPTION
**Pull request by Dee-6777**
### What type of PR is this?

_feature_

### What this PR does / Why we need it?
It scans the code and dependencies against known security vulnerabilities.

### Which Jira/Github issue(s) does this PR fix?

_Resolves_ #[OSD-17895](https://issues.redhat.com/browse/OSD-17895)  

### Special notes for your reviewer
I ran `govulncheck./...` in go version 1.21.0 and fixed the vulnerabilities. If you'll install gvulncheck and check it using some lower version of go then you might get lots of compatible issues which will eventually mess up `go.mod` and `go.sum` file. 
- [x] Ran unit tests locally
- [ ] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

**Comment by feichashao** [comment]
@Dee-6777  Thank you for the PR!
> If you'll install gvulncheck and check it using some lower version of go 
Did you mean we need to build bp-cli using go 1.21.0+ in the future?  In that case we might also bump the version in this project https://github.com/openshift/backplane-cli/blob/691eb6f0d3d4cd8a4e62a3494e1f1eed43302ca3/go.mod#L3  and make sure if run through `make test`.

**Comment by Dee-6777** [comment]
@feichashao 
> Did you mean we need to build bp-cli using go 1.21.0+ in the future? In that case we might also bump the version in this project
> 
> https://github.com/openshift/backplane-cli/blob/691eb6f0d3d4cd8a4e62a3494e1f1eed43302ca3/go.mod#L3
> 
> and make sure if run through `make test`.

Yes, we can. The thing is govulnerability depends on the system's go version. If we run(& install) govulnerability with a lower version of go it will throw version update errors of the packages, as the package version also depends on the go version. Although it's not a huge error but as we're also including CI check we have to keep this in mind.
Also if we'll include govulnvhek in ci then we have to maintain the go version of the ci along with the repo version, then only it will not throw us any error. And anyhow if the ci version is not updated it will simply throw us the previous errors and will not be fixed by simply upgrading the repo version.

**Comment by feichashao** [comment]
Yes, It seems most vulnerabilities are due to Go standard libraries.
```
govulncheck ./...

Scanning your code and 798 packages across 113 dependent modules for known vulnerabilities...

Vulnerability #1: GO-2023-1987
    Large RSA keys can cause high CPU usage in crypto/tls
  More info: https://pkg.go.dev/vuln/GO-2023-1987
  Standard library
    Found in: crypto/tls@go1.20.4
    Fixed in: crypto/tls@go1.21rc4
    Example traces found:
      #1: pkg/monitoring/monitoring.go:201:20: monitoring.Client.RunMonitoring calls http.Serve, which eventually calls tls.Conn.HandshakeContext
      #2: pkg/monitoring/monitoring.go:159:32: monitoring.Client.RunMonitoring calls io.ReadAll, which eventually calls tls.Conn.Read
      #3: pkg/utils/renderingutils.go:25:13: utils.RenderTabbedTable calls fmt.Fprintf, which calls tls.Conn.Write
      #4: pkg/utils/ocmWrapper.go:259:2: utils.DefaultOCMInterfaceImpl.IsProduction calls ocm.Connection.Close, which eventually calls tls.Dialer.DialContext

Vulnerability #2: GO-2023-1878
    Insufficient sanitization of Host header in net/http
  More info: https://pkg.go.dev/vuln/GO-2023-1878
  Standard library
    Found in: net/http@go1.20.4
    Fixed in: net/http@go1.20.6
    Example traces found:
      #1: pkg/utils/ocmWrapper.go:259:2: utils.DefaultOCMInterfaceImpl.IsProduction calls ocm.Connection.Close, which eventually calls http.Client.CloseIdleConnections
      #2: pkg/monitoring/monitoring.go:151:24: monitoring.Client.RunMonitoring calls http.Client.Do
      #3: pkg/utils/util.go:80:23: utils.CheckHealth calls http.Get
      #4: pkg/awsutil/sts.go:58:52: awsutil.AssumeRoleWithJWT calls sts.Client.AssumeRoleWithWebIdentity, which eventually calls http.Transport.CloseIdleConnections
      #5: pkg/monitoring/monitoring.go:201:20: monitoring.Client.RunMonitoring calls http.Serve, which eventually calls http.Transport.RoundTrip

Vulnerability #3: GO-2023-1840
    Unsafe behavior in setuid/setgid binaries in runtime
  More info: https://pkg.go.dev/vuln/GO-2023-1840
  Standard library
    Found in: runtime@go1.20.4
    Fixed in: runtime@go1.20.5
    Example traces found:
      #1: pkg/monitoring/monitoring.go:153:14: monitoring.Client.RunMonitoring calls log.Fatalln, which eventually calls runtime.Caller
      #2: pkg/monitoring/monitoring.go:65:44: monitoring.Client.RunMonitoring calls clientcmd.BuildConfigFromFlags, which eventually calls runtime.Callers
      #3: pkg/utils/util.go:232:49: utils.CheckBackplaneVersion calls logrus.Entry.Warn, which eventually calls runtime.CallersFrames
      #4: pkg/utils/util.go:232:49: utils.CheckBackplaneVersion calls logrus.Entry.Warn, which eventually calls runtime.Frames.Next
      #5: pkg/monitoring/monitoring.go:65:44: monitoring.Client.RunMonitoring calls clientcmd.BuildConfigFromFlags, which eventually calls runtime.Func.FileLine
      #6: pkg/monitoring/monitoring.go:156:2: monitoring.Client.RunMonitoring calls http.http2requestBody.Close, which eventually calls runtime.Func.Name
      #7: pkg/monitoring/monitoring.go:65:44: monitoring.Client.RunMonitoring calls clientcmd.BuildConfigFromFlags, which eventually calls runtime.FuncForPC
      #8: internal/upgrade/upgrade.go:76:25: upgrade.Cmd.UpgradePlugin calls fmt.Fprintln, which eventually calls runtime.GC
      #9: pkg/awsutil/credentials.go:31:33: awsutil.AWSCredentialsResponse.RenderOutput calls json.Marshal, which eventually calls runtime.GOMAXPROCS
      #10: pkg/monitoring/monitoring.go:156:2: monitoring.Client.RunMonitoring calls http.http2requestBody.Close, which eventually calls runtime.GOROOT
      #11: internal/upgrade/writer.go:35:24: upgrade.SafeWriter.Write calls os.WriteFile, which eventually calls runtime.KeepAlive
      #12: internal/upgrade/upgrade.go:76:25: upgrade.Cmd.UpgradePlugin calls fmt.Fprintln, which eventually calls runtime.ReadMemStats
      #13: pkg/cli/session/session.go:177:22: session.BackplaneSession.Start calls exec.Command, which calls runtime.SetFinalizer
      #14: pkg/cli/session/session.go:177:22: session.BackplaneSession.Start calls exec.Command, which calls runtime.Stack
      #15: cmd/ocm-backplane/login/login.go:335:23: login.doLogin calls runtime.TypeAssertionError.Error
      #16: pkg/awsutil/sts.go:15:2: awsutil.init calls sts.init, which eventually calls runtime.Version
      #17: internal/github/github.go:12:2: github.init calls runtime.init, which calls runtime.efaceOf
      #18: cmd/ocm-backplane/login/login.go:335:23: login.doLogin calls runtime.errorAddressString.Error
      #19: cmd/ocm-backplane/login/login.go:335:23: login.doLogin calls runtime.errorString.Error
      #20: internal/github/github.go:12:2: github.init calls runtime.init, which eventually calls runtime.findfunc
      #21: internal/github/github.go:12:2: github.init calls runtime.init, which calls runtime.float64frombits
      #22: internal/github/github.go:12:2: github.init calls runtime.init, which eventually calls runtime.forcegchelper
      #23: internal/github/github.go:12:2: github.init calls runtime.init, which eventually calls runtime.funcMaxSPDelta
      #24: internal/github/github.go:12:2: github.init calls runtime.init, which eventually calls runtime.lockInit
      #25: cmd/ocm-backplane/login/login.go:335:23: login.doLogin calls runtime.plainError.Error
      #26: internal/github/github.go:12:2: github.init calls runtime.init, which eventually calls runtime.throw

Your code is affected by 3 vulnerabilities from the Go standard library.

Share feedback at https://go.dev/s/govulncheck-feedback.

```

File: Makefile [hunk 1]
```diff
@@ -121,6 +121,10 @@ build-image:
 lint-in-container: build-image
 	$(RUN_IN_CONTAINER_CMD) "go mod download && make lint"
 
+.PHONY: scan
+scan: 
+	govulncheck./...
```
Comment (feichashao) [inline Makefile:126]:
```suggestion
	govulncheck ./...
```

**Comment by Dee-6777** [comment]
> Yes, It seems most vulnerabilities are due to Go standard libraries.
@feichashao So, do we need to bump the go version now?

**Comment by feichashao** [comment]
@Dee-6777 for bumping the go version, I would suggest open another PR to update `go.mod`.

**Pull request merged**